tests/test_all_fields.py

```diff
@@ -43,43 +90,32 @@ def test_run(self):
         for stream in expected_streams:
             with self.subTest(stream=stream):
 
-                # expected values
-                expected_automatic_keys = self.expected_primary_keys()[stream] | self.expected_replication_keys()[stream]
+                # Expected values
+                expected_automatic_keys = self.expected_primary_keys()[stream] | \
+                    self.expected_replication_keys()[stream]
 
-                # get all expected keys
+                # Get all expected keys
                 expected_all_keys = stream_to_all_catalog_fields[stream]
 
-                # collect actual values
+                # Collect actual values
                 messages = synced_records.get(stream)
 
                 actual_all_keys = set()
-                # collect actual values
+                # Collect actual values
                 for message in messages['messages']:
                     if message['action'] == 'upsert':
                         actual_all_keys.update(message['data'].keys())
 
                 # Verify that you get some records for each stream
                 self.assertGreater(record_count_by_stream.get(stream, -1), 0)
```

Comment:
Is this validation really needed here? We have verified it in other test cases and this one is not specific to all fields testing.

Reply:
I would prefer it there because this makes sure that we have at least one record before we do other validations, Lack of test data can lead to assertions.